mcs/restapi/admin_groups.go

```diff
@@ -166,3 +217,114 @@ func getRemoveGroupResponse(params admin_api.RemoveGroupParams) error {
 	}
 	return nil
 }
+
+// updateGroup updates a group by adding/removing members and setting the status to the desired one
+//
+// isRemove: whether remove members or not
+func updateGroupMembers(client MinioAdmin, group string, members []string, isRemove bool) error {
+	gAddRemove := madmin.GroupAddRemove{
+		Group:    group,
+		Members:  members,
+		IsRemove: isRemove,
+	}
+	err := client.updateGroupMembers(gAddRemove)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// addOrDeleteMembers updates a group members by adding or deleting them based on the expectedMembers
+func addOrDeleteMembers(client MinioAdmin, group *madmin.GroupDesc, expectedMembers []string) error {
+	// get members to delete/add
+	membersToDelete := DifferenceArrays(group.Members, expectedMembers)
+	membersToAdd := DifferenceArrays(expectedMembers, group.Members)
+	// delete members if any to be deleted
+	if len(membersToDelete) > 0 {
+		err := updateGroupMembers(client, group.Name, membersToDelete, true)
+		if err != nil {
+			return err
+		}
+	}
+	// add members if any to be added
+	if len(membersToAdd) > 0 {
+		err := updateGroupMembers(client, group.Name, membersToAdd, false)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func setGroupStatus(client MinioAdmin, group, status string) error {
+	var setStatus madmin.GroupStatus
+	switch status {
+	case "enabled":
```

Comment:
this is a good opportunity for a constant or an enum

Reply:
yes I thought about that mhmh but not sure if restrict it since minio might change the status

Reply:
will do if necessary in future